src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php

```diff
@@ -211,4 +202,72 @@ private function checkResourceLocator()
         return !($this->requestAnalyzer->getResourceLocator() === '/'
             && $this->requestAnalyzer->getResourceLocatorPrefix());
     }
+
+    /**
+     * @param Request $request
+     * @return Route
+     */
+    protected function getRedirectWebSpaceRoute(Request $request)
+    {
+        // redirect by information from webspace config
+        return new Route(
+            $request->getPathInfo(), array(
+                '_controller' => 'SuluWebsiteBundle:Default:redirectWebspace',
+                'url' => $this->requestAnalyzer->getPortalUrl(),
+                'redirect' => $this->requestAnalyzer->getRedirect()
+            )
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @param $url
+     * @return Route
+     */
+    protected function getRedirectRoute(Request $request, $url)
+    {
+        // redirect to linked page
+        return new Route(
+            $request->getPathInfo(), array(
+                '_controller' => 'SuluWebsiteBundle:Default:redirect',
+                'url' => $url
+            )
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @param StructureInterface $content
+     * @return Route
+     */
+    protected function getStructureRoute(Request $request, $content)
+    {
+        return new Route(
+            $request->getPathInfo(), array(
+                '_controller' => $content->getController(),
+                'structure' => $content
+            )
+        );
+    }
+
+    /**
+     * @param $resourceLocator
+     * @return string
+     */
+    protected function getUrlWithoutFormat($resourceLocator)
```

Comment:
Might be, but this method still doesn't remove any format, it looks like it's only removing a trailing slash.